ui/js/components/perf/compare.js

```diff
@@ -108,3 +108,52 @@ treeherder.component('compareError', {
         newRevision: '<'
     }
 });
+
+treeherder.component('distributionGraph', {
+    template: '<canvas id="a" width="400" height="30"></canvas>',
+    bindings: {
+        replicates: '<'
+    },
+    controller: [function () {
+        var ctrl = this;
+        var cvs = document.getElementById("a");
+        var ctx = cvs.getContext("2d");
+        var max = Math.ceil(Math.max.apply(null, ctrl.replicates)*1.01);
+        var min = Math.floor(Math.min.apply(null, ctrl.replicates)/1.01);
+        //draw borders
+        ctx.strokeStyle = 'white';
+        ctx.strokeRect(0, 0, 400, 30);
+        ctx.beginPath();
+        ctx.moveTo(50, 0);
+        ctx.lineTo(50, 30);
+        ctx.stroke();
+        ctx.beginPath();
+        ctx.moveTo(350, 0);
+        ctx.lineTo(350, 30);
+        ctx.stroke();
+        //write min and max value for the range
+        ctx.font = '13px helvetica';
+        ctx.textAlign = 'center';
+        ctx.fillStyle = 'white';
+
+        if (min <= 999999)
+            ctx.fillText(min.toFixed(0), 25, 20);
+        else
+            ctx.fillText(numeral(min).format('0.00a'), 25, 20);
+
+        if (max <= 999999)
+            ctx.fillText(max.toFixed(0), 375, 20);
+        else
+            ctx.fillText(numeral(max).format('0.00a'), 25, 20);
+
+        //draw data points
+        ctx.globalAlpha = 0.3;
+        _.each(ctrl.replicates, function (element) {
```

Comment:
Probably better to use ctrl.replicates.forEach here. Also, you can use a new-style arrow function which is a little less verbose (http://airbnb.io/javascript/#arrows--use-them):

```js
ctrl.replicates.forEach((value) => {
    ....
}
```